polymorphia-backend/src/main/java/com/agh/polymorphia_backend/service/course/CourseImportService.java

```
@@ -0,0 +1,347 @@
+                .build();
+        updateCourseDetails(request, course);
+        Course savedCourse = courseRepository.save(course);
+        courseRepository.flush();
+        Long courseId = savedCourse.getId();
+
+        UserCourseRole userCourseRole = UserCourseRole.builder()
+                .course(savedCourse)
+                .user(currentUser)
+                .id(new UserCourseRoleId(currentUser.getId(), courseId))
+                .role(UserType.COORDINATOR)
+                .build();
+        userCourseRoleRepository.save(userCourseRole);
+
+        updateCourseElements(request, new CourseDetailsRequestDto(), courseId);
+
+    }
+
+    @Transactional
+    public void updateCourse(CourseDetailsRequestDto request, Long courseId) {
+        CourseDetailsRequestDto currentConfig = courseDetailsService.getCourseDetails(courseId);
+
+        if (!request.equals(currentConfig)) {
+            Course course = courseService.getCourseById(courseId);
+            updateCourseDetails(request, course);
+            courseRepository.save(course);
+        }
+        updateCourseElements(request, currentConfig, courseId);
+    }
+
+    private void updateCourseElements(CourseDetailsRequestDto request, CourseDetailsRequestDto currentConfig, Long courseId) {
+        updateEvolutionStages(request.getEvolutionStages(), currentConfig.getEvolutionStages(), courseId);
+        updateEventSections(request.getEventSections(), currentConfig.getEventSections(), courseId);
+        updateItems(request.getItems(), currentConfig.getItems(), courseId);
+        updateChests(request.getChests(), currentConfig.getChests(), courseId);
+        updateAllCriteriaRewards(request.getEventSections(), courseId);
+        updateRoadmapOrder(request.getRoadmapOrderKeys(), courseId);
+    }
+
+    private void updateRoadmapOrder(List<String> roadmapOrderKeys, Long courseId) {
+        if (roadmapOrderKeys == null || roadmapOrderKeys.isEmpty()) {
+            return;
```

Maybe we should throw an error if it's null?

Also, I know that we want to limit validation here but maybe we should check if all gradable events that belong in event sections with `isShownInRoadmap = true` are in roadmapOrderKeys and there are no duplicates? It's not the highest priority though.